docker/grafana/Dockerfile

```diff
@@ -14,8 +14,7 @@ VOLUME /dashboards
 ENV GF_SECURITY_ADMIN_USER=admin
 ENV GF_SECURITY_ADMIN_PASSWORD=admin
 
-# Modified datasource to work with a network_mode: host
-ENV PROMETHEUS_URL=http://prometheus:9090
```

Comment:
Do we need to override those here? Might be better to just do it in the compose file where `extra_hosts` is set

Reply:
Unclear to me why it was done like that in the first place. Maybe there are other ways those Dockerfile are used?
To reduce potential for regression I'd rather keep this PR focused on adding OSX support and not introduce other refactoring.

Reply:
> Maybe there are other ways those Dockerfile are used?

Yes, we publish those to docker hub, users might rely on host being set to `prometheus` here